src/App.tsx

```diff
@@ -15,36 +27,118 @@
   'Garlic',
 ];
 
+const goodsWithKeys = goodsFromServer.map((good, i) => {
+  return { id: i + 1, product: good };
+});
+
+type Action =
+  | { type: SortType.Alphabetical }
+  | { type: SortType.Length }
+  | { type: SortType.Reverse }
+  | { type: SortType.Clear };
+
+type State = {
+  sortedList: ProductWithKey[];
+  sortedBy: string;
+  isReversed: boolean;
+};
+
 export const App: React.FC = () => {
+  const initialState = {
+    sortedList: [...goodsWithKeys],
+    sortedBy: '',
+    isReversed: false,
+    isChanged: false,
+  };
+
+  function reducer(state: State, action: Action): State {
```

Comment:
[CODE STYLE] - The reducer function's name 'reducer' is not descriptive of its purpose. Consider renaming it to something more indicative of its function, such as 'sortListReducer', to improve code readability.